### DISCUSSION TITLE
bcftools error

### DISCUSSION THREAD

**Discussion by Jay-Leung** [biostars/biostar-handbook-issues#201] [Q&A] [answered]
Hello! I was following the step by step installation from [https://www.biostarhandbook.com/step-by-step.html#step-by-step](url).

However, when I ran `./bin/doctor.py`, it gave an error for bcftools:

Doctor! Doctor! Give me the news.
Checking symptoms ...
bwa ... OK
datamash ... OK
fastqc -h ... OK
hisat2 ... OK
seqret -h ... OK
featureCounts ... OK
efetch ... OK
esearch ... OK
samtools ... OK
fastq-dump ... OK
bowtie2 ... OK
bcftools ... ERROR
seqtk ... OK
seqkit ... OK
bio ... OK
fastq-dump -X 1 -Z SRR1553591 ... OK

 Your system shows 1 error!

Run the command that has failed for more details.

When I ran `bcftools`, it gave:
bcftools: error while loading shared libraries: libgsl.so.25: cannot open shared object file: No such file or directory

Could you advise on this? I am using a Linux Ubuntu LTS 20.04 LTS virtual machine. I have tried to install the latest gsl 2.7 but it doesn't seem to work as well.

Thanks!

**Comment by ialbert** [comment]
sounds like some sort of library version mismatch, sometimes this can be fixed by forcing either the tool or the library to the correct version:

`mamba install bcftools=1.4`

or , since this looks like it is looking for an older version of gsl (25 most likely means 2.5)

`mamba install gsl=2.5
`

another possibility is to install `bcftools` into a separate environment and switch to that when running `bcftools`

**Comment by Jay-Leung** [comment, marked as answer]
Hi Albert!

Thanks for the prompt response! I installed gsl 25 and bcftools works, but samtools did not. I resolved it by running `conda update --all`, so now all is okay. Thanks!

Regards,
Jay

**Reply by ialbert** [reply]
sound good, glad it worked out
